overlord/managers_test.go

```diff
@@ -3734,6 +3850,151 @@ version: 1.0`
 	}
 }
 
+func (ms *mgrsSuite) TestRemodelSwitchToDifferentKernelUndo(c *C) {
```

Comment:
we still need a failover test right ?

Reply:
It will look essentially the same AFAICT, the error will happen in "doAutoConnect" in this case. So  I could ensure the error trigger task is running right after this.

Reply:
not really, the boot env state would evolve differently, also as we discussed we should really make the tests more realistic and test the ensureBootOk bits as well by having some kind of reset for those

Reply:
I added a rollback scenario now too